src/Proxy.hs

```diff
@@ -890,5 +894,36 @@ putGTUDropR addrText = do
 getGlobalFileR :: Handler TypedContent
 getGlobalFileR = toTypedContent . globalInfo <$> getYesod
 
+-- Queries the transaction database and the GRPC, 
+-- then if both succeed checks that the last final block is less than 5 minutes old.
+getHealthR :: Handler TypedContent
+getHealthR = 
+  runGRPC doGetBlockFinalInfo $ \case
+      Nothing -> do
+        $(logError) $ "Could not get response from GRPC."
+        sendResponse $ object $ ["healthy" .= False, "reason" .= ("Could not get response from GRPC.":: String)]
+      Just lastFinalBlockInfo -> do
+        $(logInfo) "Successfully got best block info."
+        _ :: [(Entity Entry, Entity Summary)] <- runDB $ E.select $
+                  E.from $ \val -> do
+                  E.limit 1
+                  return val
+        -- get block slot time from block info object, compare with current time: reject if more than 5 minutes
+        case fromJSON lastFinalBlockInfo of
+          Error _ -> sendResponse $ object $ ["healthy" .= False, "reason" .= ("Block info format has changed. blockSlotTime is not UTCTime.":: String)]
```

Comment:
I suppose the error message should change to reflect the change in parsing.

I think in this case a better response would be 
```haskell
sendResponseStatus badGateway502 $ object [
        "errorMessage" .= i18n i EMGRPCError,
        "error" .= fromEnum InternalError
        ]
```

since this is an internal server error, not something that has directly to do with health.